### ISSUE TITLE
salt.beacons.inotify not sending event

### ISSUE THREAD

**Issue by indigoarc** [saltstack/salt#47812]
### Description of Issue/Question
Followed the example in salt.beacons.inotify, but not seeing any event on salt master

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
1. Add inotify configuration under /etc/salt/minion.d./
[root@xxx minion.d]#cat beacons.conf 
beacons:
  inotify:
    - files:
        /etc/important_file:
          mask:
            - modify
    - disable_during_state_run: True
2. create a file to monitor
[root@xxx minion.d]#cat /etc/important_file 
afsaiasfsffafasfs

3. restart minion service

4. on master, run "salt-run state.event pretty=true"

5. on minion, modify and save the important_file content. 

6. Expect to see event coming on master, but nothing generated.

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Mar 30 2018, 15:31:59)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.9 Final
         locale: UTF-8
        machine: x86_64
        release: 2.6.32-696.23.1.el6.centos.plus.x86_64
         system: Linux
        version: CentOS 6.9 Final

**Comment by garethgreenaway** [comment]
@indigoarc Thanks for the report.  Unfortunately I was unable to reproduce this issue, this is the event that I see on the event bus when I modify the file in question:
```
salt/beacon/blackoil/inotify//etc/important_file        {
    "_stamp": "2018-05-24T16:10:12.898827",
    "change": "IN_IGNORED",
    "id": "blackoil",
    "path": "/etc/important_file"
}
```
I see this event when I am modifying the contents of an existing file.  Have you tried other inotify mask options to see if you receive events?

**Comment by indigoarc** [comment]
I tried different options and none of them works. Tried to monitor the whole directory also like below.

[root@minion1 etc]#cat /etc/salt/minion.d/beacons.conf 
beacons:
  inotify:
    - files:
        /etc:
          mask:
            - modify
            - open
            - delete
            - open
            - close_write
            - access 
            - attrib
            - close_nowrite 
            - close_write 
            - create 
            - delete_self
            - moved_from 
            - moved_to
            - move_self
          recurse: True
          auto_add: True
    - disable_during_state_run: True
    - coalesce: True

**Comment by indigoarc** [comment]
So which dependency package should I install?
According to this tutorial
 [https://docs.saltstack.com/en/getstarted/event/beacons.html](url)
I should install python-pyinotify which failed to install.

But according another document,
[https://docs.saltstack.com/en/latest/topics/beacons/index.html](url)
I should install python-inotify which installed fine.

**Comment by garethgreenaway** [comment]
You want `python-pynotify` if it's not installed that would explain why it's not firing events as it's not available.  How does it fail to install?.

**Comment by indigoarc** [comment]
```
[root@master ~]#salt "*hs1*" pkg.install python-pyinotify
hs1:
    ERROR: Error occurred installing package(s). Additional info follows:
    
    changes:
        ----------
    errors:
        - Loaded plugins: fastestmirror, refresh-packagekit, rhnplugin, security
          This system is receiving updates from RHN Classic or Red Hat Satellite.
          Setting up Install Process
          Loading mirror speeds from cached hostfile
           * epel: mirror.vcu.edu
          No package python-pyinotify available.
          Error: Nothing to do
```
How can I get this pkg installed?
Thanks!

**Comment by garethgreenaway** [comment]
The package you want for CentOS is python-inotify which in 6.x is version 0.9.1, the beacon needs version 0.9.5 or higher for some functionality.  It's likely that the version you had installed does not include the required functionality and so the beacon is not loading.  Alternatively you could install the Python library via the `pip` installer.

**Comment by indigoarc** [comment]
Yes, problem solved with the following states file to install the correct Python library

```
install_python:
  pkg.installed:
    - name: python27-pip

install_inotify:
  pip.installed:
    - name: pyinotify
```
The online document should be updated. Thanks for your help!
